gradle-plugin/src/main/java/com/google/cloud/tools/dependencies/gradle/LinkageCheckTask.java

```diff
@@ -307,14 +309,20 @@ private static DependencyGraph createDependencyGraph(ResolvedConfiguration confi
 
       DependencyPath parentPath = item.getParentPath();
 
-      // parentPath is null for the first item
-      DependencyPath path =
-          parentPath == null
-              ? new DependencyPath(artifactFrom(node))
-              : parentPath.append(dependencyFrom(node));
+      DependencyPath path = null;
 
-      graph.addPath(path);
+      // For artifacts with classifiers, there can be multiple resolved artifacts for one node
+      for (ResolvedArtifact artifact : node.getModuleArtifacts()) {
+        // parentPath is null for the first item
+        path =
+            parentPath == null
+                ? new DependencyPath(artifactFrom(node, artifact))
+                : parentPath.append(dependencyFrom(node, artifact));
+        graph.addPath(path);
+      }
 
+      Verify.verify(
```

Comment:
Is `path != null` equivalent to `!node.getModuleArtifacts().isEmpty()`? If so you may be able to just verify that condition instead which would allow you to avoid defining `path` outside the loop.

Reply:
Updated this to check more earlier.